LICENSE

```diff
@@ -210,4 +210,38 @@ for text of these licenses.
 Apache License 2.0
 --------------------------------------
 
-common/src/main/java/org/apache/celeborn/common/network/util/LimitedInputStream.java
+Apache Spark
+./client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/SparkUtils.java
+./client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SparkUtils.java
+./worker/src/main/java/org/apache/celeborn/service/deploy/worker/shuffledb/DB.java
+./worker/src/main/java/org/apache/celeborn/service/deploy/worker/shuffledb/DBIterator.java
+./worker/src/main/java/org/apache/celeborn/service/deploy/worker/shuffledb/LevelDB.java
+./worker/src/main/java/org/apache/celeborn/service/deploy/worker/shuffledb/LevelDBProvider.java
+./worker/src/main/java/org/apache/celeborn/service/deploy/worker/shuffledb/RocksDBIterator.java
+./worker/src/main/java/org/apache/celeborn/service/deploy/worker/shuffledb/StoreVersion.java
+./worker/src/main/java/org/apache/celeborn/service/deploy/worker/shuffledb/DBBackend.java
+./worker/src/main/java/org/apache/celeborn/service/deploy/worker/shuffledb/DBProvider.java
+./worker/src/main/java/org/apache/celeborn/service/deploy/worker/shuffledb/LevelDBIterator.java
+./worker/src/main/java/org/apache/celeborn/service/deploy/worker/shuffledb/RocksDB.java
+./worker/src/main/java/org/apache/celeborn/service/deploy/worker/shuffledb/RocksDBProvider.java
+
+Apache Kyuubi
+./common/src/main/java/org/apache/celeborn/reflect/DynClasses.java
+./common/src/main/java/org/apache/celeborn/reflect/DynConstructors.java
+./common/src/main/java/org/apache/celeborn/reflect/DynFields.java
+./common/src/main/java/org/apache/celeborn/reflect/DynMethods.java
+./dev/checkout_pr.sh
```

Comment:
I think this is redundant, the author of the file belongs to the same person.

Reply:
https://github.com/apache/iceberg/tree/22d4e7836ac0bb47bed627e7c773d44105b8f0f8/common/src/main/java/org/apache/iceberg/common    
Sorry, I checked the source of the code again, these codes should come from iceberg, not Kyuubi. at least that's how it looks on the timeline.

Reply:
AFAIK, the `Dyn*` classes originated from the Apache Parquet project, then 
were borrowed by Apache Iceberg, Apache Kyuubi, and I copied them to Apache Celeborn (incubating). Specifically, I copied a modified version of the Kyuubi project. How should I write the NOTICE or LICENSE is correct then?

I had thought that copying code across ASF projects did not require listing files under LICENSE, because, legally speaking, the copyright owner of the code is ASF, not some natural persons.

> I think this is redundant, the author of the file belongs to the same person.

Sorry, I don't know such a rule. Could you clarify where I can find such guidance that clearly states that the same author could legally contribute the same code to different projects without declarations, even though the copyright of the code belongs to different organizations or companies?

Let's say Tom contributes some code to a project under the GPL license, and the copyright owner is company A. Could Tom legally contribute the same code to an ASF project, without any declarations?

I think all code contributed to the ASF projects belongs to ASF, there is no difference between different contributors.

TBH, different people have different interpretations of LICENSE rules, I believe that I and other community members always respond positively when someone, especially someone we consider to be an expert in the field, adapts to the guidance.

Reply:
Listing the files in LICENSE is a good idea so users know where they come from. Is it not a requirement, but anything that makes the licenses easier to check is good by me. If you copied them from Apache Kyuubi, I would use their NOTICE file contents.

Reply:
@justinmclean thanks for your clarification.

In the past, when borrowing code from external projects(non-ASF projects) to an ASF project, I would carefully evaluate the LICENSE and list the copied files under LICENSE and NOTICE. In the future, I will follow your guidance to list files copied from ASF projects too.

Reply:
Note that third-party Apache Licensed files will have different headers. All ASF project files have the same header, so it is hard to know where a given file comes from.

Reply:
Yeah, I know the LICENSE header's rule, it's documented under the ASF website clearly.

Reply:
> it is hard to know where a given file comes from.

we usually document where the code comes from in the class javadocs :)

Reply:
Apache products are composed of lots of pieces of code across numerous source files, licensed to the ASF by various authors who maintain ownership of their contributions.

In short, I contributed a piece of code to Skywalking, but I still own the code, so I can feel free to contribute the code to other projects, even if its license is incompatible with AL2. Because the ownership is mine.

Reply:
@CalvinKirs would you mind providing the legal basis for your statement? especially are those provisions documented under the Apache official site? I may miss something.

Reply:
@justinmclean, I have started new vote thread: [[VOTE] Release Apache Celeborn(Incubating) 0.3.2-incubating-rc2](https://lists.apache.org/thread/4zks008fyqo5om6tg0872xbz9wt562ds). If you are available, please help to check this rc.